Make the EHCI host driver respect the "nousb" kernel option.

Signed-off-by: Chuck Ebbert <cebbert@redhat.com>

---
 drivers/usb/host/ehci-hcd.c |    3 +++
 1 file changed, 3 insertions(+)

--- linux-2.6.23.noarch.orig/drivers/usb/host/ehci-hcd.c
+++ linux-2.6.23.noarch/drivers/usb/host/ehci-hcd.c
@@ -973,6 +973,9 @@ static int __init ehci_hcd_init(void)
 {
 	int retval = 0;
 
+	if (usb_disabled())
+		return -ENODEV;
+
 	pr_debug("%s: block sizes: qh %Zd qtd %Zd itd %Zd sitd %Zd\n",
 		 hcd_name,
 		 sizeof(struct ehci_qh), sizeof(struct ehci_qtd),
